packages/server/src/errors.ts

```diff
@@ -0,0 +1,15 @@
+export class HttpError extends Error {
+  code: number;
+  error?: unknown;
+
+  constructor(code: number, message: string, error?: unknown) {
```

Comment:
How would you deal with business logic error codes?

Reply:
It will be another class extends `Error`, so we can capture them during the same error handling function and do some custom handling for business logic error.
For each such error, base on the context, we may give a specific http error code or just 200. But it will have another field `errorCode` for storing the business error code. The response should be something similar to `{ errorCode: number; reason: string; }`